systemtest/processor_test.go

```diff
@@ -430,3 +431,79 @@ func TestCallbackFail(t *testing.T) {
 	log.Printf("%v", err)
 	test.AssertTrue(t, strings.Contains(err.Error(), "callback failed"))
 }
+
+func TestMessageCommit(t *testing.T) {
+
+	var (
+		group       goka.Group  = goka.Group(fmt.Sprintf("%s-%d", "goka-systemtest-message-commit", time.Now().Unix()))
+		inputStream goka.Stream = goka.Stream(group) + "-input"
+	)
+
+	if !*systemtest {
+		t.Skipf("Ignoring systemtest. pass '-args -systemtest' to `go test` to include them")
+	}
+
+	emitter, err := goka.NewEmitter([]string{*broker}, inputStream, new(codec.Int64))
+	test.AssertNil(t, err)
+
+	tmc := goka.NewTopicManagerConfig()
+	tmc.Table.Replication = 1
+	tmc.Stream.Replication = 1
+	cfg := goka.DefaultConfig()
+	// we want to consume all messages
+	cfg.Consumer.Offsets.Initial = sarama.OffsetOldest
+	goka.ReplaceGlobalConfig(cfg)
+
+	tmBuilder := goka.TopicManagerBuilderWithConfig(cfg, tmc)
+	tm, err := tmBuilder([]string{*broker})
+	test.AssertNil(t, err)
+
+	tm.EnsureStreamExists(string(inputStream), 10)
+
+	time.Sleep(10 * time.Second)
+
+	log.Printf("emitting 10 messages")
+	for i := 0; i < 10; i++ {
+		emitter.EmitSync("1", int64(1))
+	}
+	// close emitter
+	test.AssertNil(t, emitter.Finish())
+
+	for i := 0; i < 3; i++ {
+		log.Printf("starting processor")
+		done := make(chan struct{})
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		proc, err := goka.NewProcessor([]string{*broker}, goka.DefineGroup(group,
+			goka.Input(inputStream, new(codec.Int64), func(ctx goka.Context, msg interface{}) {
+				log.Printf("consuming message")
+				if val := ctx.Value(); val == nil {
+					ctx.SetValue(msg)
+				} else {
+					ctx.SetValue(val.(int64) + msg.(int64))
+				}
+			}),
+			goka.Persist(new(codec.Int64)),
+		),
+			goka.WithTopicManagerBuilder(tmBuilder),
+		)
+		test.AssertNil(t, err)
+		// run a new processor
+		go func() {
+			defer close(done)
+			err := proc.Run(ctx)
+			test.AssertNil(t, err)
+		}()
+
+		time.Sleep(10 * time.Second)
+		val, err := proc.Get("1")
+		test.AssertNil(t, err)
+		test.AssertTrue(t, val != nil)
+		test.AssertEqual(t, val.(int64), int64(10))
+		log.Printf("received 10 messages")
```

Comment:
Sorry, but i don't know how this tests the commit of the input topic. 
It looks like it only tests the `ctx.SetValue`, but not the commit of the input topic. 
Maybe i miss something?

Reply:
sorry this lacks some description, was a bit in a hurry :) So basically the processor just receives and accumulates the value. There's only one key. Now the idea was that we restart the same processor a couple of times and check the value of the one key we ever store. It should never change, because the first processor run processes all messages and there aren't any more.
Without the "+1"-fix, it keeps increasing with each restart of the processor (and the test fails), as the last message is being reprocessed every time, thus incrementing the persisted value.
I'll add something like this as a comment.

Reply:
Right. Thanks for the explanation 👍🏼